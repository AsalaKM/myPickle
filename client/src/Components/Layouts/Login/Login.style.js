import styled from "styled-components"

export const LoginPage = styled.div`
  display: flex;
  flex-direction: column;
  width: 80%;
  margin: 0 auto;
  margin-top: 125px;
`
export const HomeIcon = styled.img`
  height: 30px;
  width: 60px;
  margin-top: -99px;
  margin-left: -30px;
`

export const Input = styled.input`
  height: 50px;
  width: 100%;
  border: none;
  border-radius: 9px;
  background: #f8fbfa;
  font-size: 18px;
  text-align: center;
  display: block;
`

export const Logo = styled.img`
  width: 114px;
  margin-top: 26px;
  height: 185px;
`

export const Title = styled.h2`
  text-align: center;
  display: block;
`
export const Button = styled.button`
  height: 50px;
  width: 100%;
  border: none;
  border-radius: 5px;
  background: #6eeb26;
`

export const RegisterLink = styled.h2`
  display: flex;
  justify-content: center;
  align-items: center;
`

export const Branding = styled.div`
  flex-direction: column;
  display: flex;
  justify-content: center;
  align-items: center;
`
