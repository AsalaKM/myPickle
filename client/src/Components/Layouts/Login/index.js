import React, { Component } from "react"
import axios from "axios"

import home from "../../../assets/images/home.svg"
import logo from "../../../assets/images/logo.jpg"

import {
  LoginPage,
  HomeIcon,
  Logo,
  Title,
  Input,
  Button,
  RegisterLink,
  Branding,
} from "./Login.style.js"

import { ErrorMsg } from "../../Common/Questions/Questions.style"

class Login extends Component {
  state = {
    email: "",
    password: "",
    errors: {},
  }
  onChange = e => {
    this.setState({
      [e.target.name]: e.target.value,
    })
  }

  clickHome = () => {
    this.props.history.push("/")
  }

  clickResgister = () => {
    this.props.history.push("/register")
  }

  handleSubmit = e => {
    e.preventDefault()
    const { email, password } = this.state
    const user = { email: email, password: password }

    axios
      .post("/login-user", user)
      .then(res => console.log("result", res))
      .catch(err => this.setState({ errors: err.response.data }))
  }

  render() {
    const { errors } = this.state
    return (
      <LoginPage>
        <HomeIcon src={home} onClick={this.clickHome} />

        <Branding>
          <Logo src={logo} />
          <Title>myPikle</Title>
        </Branding>

        <form>
          <Input
            type="email"
            placeholder="Your Email"
            name="email"
            value={this.state.email}
            onChange={this.onChange}
          />
          {errors.email && <ErrorMsg className="invalid-feedback">{errors.email}</ErrorMsg>}
          <br />
          <Input
            type="password"
            placeholder="Password"
            name="password"
            value={this.state.password}
            onChange={this.onChange}
          />
          {errors.password && <ErrorMsg className="invalid-feedback">{errors.password}</ErrorMsg>}
          <br />
        </form>

        <br />

        <Button onClick={this.handleSubmit}>Login</Button>

        <RegisterLink onClick={this.clickResgister}>Create new acccount!</RegisterLink>
      </LoginPage>
    )
  }
}

export default Login
