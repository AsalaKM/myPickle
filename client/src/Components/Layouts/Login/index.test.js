import React from "react"
import renderer from "react-test-renderer"
import { StaticRouter } from "react-router-dom"

import Login from "./index"

test("Login matches snapshot", () => {
  const context = {}
  const component = renderer.create(
    <StaticRouter context={context}>
      <Login />
    </StaticRouter>
  )
  expect(component).toMatchSnapshot()
})
