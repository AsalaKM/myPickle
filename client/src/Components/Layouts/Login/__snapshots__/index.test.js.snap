// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
<div
  className="sc-bdVaJa caUQkH"
>
  <img
    className="sc-bwzfXH fesAsK"
    onClick={[Function]}
    src="home.svg"
  />
  <div
    className="sc-gzVnrw kpZGLP"
  >
    <img
      className="sc-bxivhb hTUQbg"
      src="logo.jpg"
    />
    <h2
      className="sc-ifAKCX Tiktc"
    >
      myPikle
    </h2>
  </div>
  <form>
    <input
      className="sc-htpNat hfCgQd"
      name="email"
      onChange={[Function]}
      placeholder="Your Email"
      type="email"
      value=""
    />
    <br />
    <input
      className="sc-htpNat hfCgQd"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <br />
  </form>
  <br />
  <button
    className="sc-EHOje iUMJbr"
    onClick={[Function]}
  >
    Login
  </button>
  <h2
    className="sc-bZQynM eHnwNe"
    onClick={[Function]}
  >
    Create new acccount!
  </h2>
</div>
`;
